File: backend/app/api/questions.py
from fastapi import APIRouter, Query

from backend.app.schemas.question import Question
from backend.app.schemas.response import ApiResponse
from backend.app.services.question_service import get_today_questions

router = APIRouter()


@router.get(
    "/questions/today",
    response_model=ApiResponse[list[Question]],
    summary="获取今天要练的题",
    description="用于获取今天要练的题目列表",
)
async def today_questions(
        limit: int = Query(default=5, ge=1, le=10),
):
    questions = get_today_questions(limit)

    return ApiResponse(data=questions)

File: frontend/src/api/questions.ts
/**
 * 文件作用：早期“今日题目” API 封装；当前主线正逐步迁移到 practice.ts 的真实练习接口。
 */

import {http} from '@/lib/http';
import type {ApiResponse} from "@/types/api.ts"

export type Question = {
    id: number
    question: string
    topic: string
}

export const getTodayQuestions = (limit = 5) => {
    return http
        .get<ApiResponse<Question[]>>("/questions/today", {
            params: {limit},
        })
        .then((res) => res.data)
}

